include/k4a/k4atypes.h

```diff
@@ -296,6 +299,17 @@ typedef enum
     K4A_DEPTH_MODE_PASSIVE_IR,     /**< Passive IR only, captured at 1024x1024. */
 } k4a_depth_mode_t;
 
+/** Depth sensor capture mode strings.
+ *
+ * \remarks
+ * If strings are added, update the k4a_depth_mode_t enum.
+ */
+static const char *k4a_depth_mode_strings[6] = {
```

Comment:
I'm not sure about adding this to the public API surface. This may be more appropriate as an internal helper.

There's no way to easily know the bounds of a value that is being used to lookup in to this array, so it would be pretty easy for someone to translate k4a_depth_modes_strings[modevariable] without verifying that modevariable is between 0 and 5.

It might be better to have an internal helper function like this:

```C
#define K4A_DEPTH_MODE_TO_STRING_CASE( mode ) case mode: return #mode
const char* k4a_depth_mode_to_string(k4a_depth_mode_t mode)
{
   switch (mode)
   {
       K4A_DEPTH_MODE_TO_STRING_CASE( K4A_DEPTH_MODE_OFF );
       K4A_DEPTH_MODE_TO_STRING_CASE( K4A_DEPTH_MODE_NFOV_UNBINNED );
       K4A_DEPTH_MODE_TO_STRING_CASE( K4A_DEPTH_MODE_WFOV_UNBINNED );
       etc.
   }
   return "unexpected k4a_depth_mode_t value";
}
```

That function will ensure that the inputs are sanitized, and that the strings always match the enum names or it will be a build break. I think with high warning levels, clang may provide a build break if a new enum value is added and not all the switch cases are covered.